### PR TITLE
feat(sizing): Use the sizable mixins and functions to size components.

### LINKED ISSUE #12347: [Themes] - Use the sizable mixins and functions to size components
Use the theming package when sizing components in place of the current approach.

### PR DESCRIPTION
**Pull request by desig9stein**
Closes #12347 

### Additional information (check all that apply):
 - [ ] Bug fix
 - [ ] New functionality
 - [ ] Documentation
 - [ ] Demos
 - [ ] CI/CD

### Checklist:
 - [ ] All relevant tags have been applied to this PR
 - [ ] This PR includes unit tests covering all the new code ([test guidelines](https://github.com/IgniteUI/igniteui-angular/wiki/Test-implementation-guidelines-for-Ignite-UI-for-Angular))
 - [ ] This PR includes API docs for newly added methods/properties ([api docs guidelines](https://github.com/IgniteUI/igniteui-angular/wiki/Documentation-Guidelines))
 - [ ] This PR includes `feature/README.MD` updates for the feature docs
 - [ ] This PR includes general feature table updates in the root `README.MD`
 - [ ] This PR includes `CHANGELOG.MD` updates for newly added functionality
 - [ ] This PR contains breaking changes
 - [ ] This PR includes `ng update` migrations for the breaking changes ([migrations guidelines](https://github.com/IgniteUI/igniteui-angular/wiki/Update-Migrations))
 - [ ] This PR includes behavioral changes and the feature specification has been updated with them

### CODE REVIEW

**Comment by SisIvanova** [comment]
Some differences against the master branch:
- Avatar Initials - Font-size
- Badge Size - Test it with any random four digits 
- Banner Padding
- Dropdown Groups - Left Indent 
- Grid Excel Style Filtering Padding - See the Grid External Filtering Sample
- Empty Grid Display Container - See the attachment
Screenshot 2023-04-24 at 1 39 55 PM

**Comment by SisIvanova** [comment]
All of the previously mentioned issues are fixed. However, I found one other problem related to the dropdown component. 
The padding of the `igx-combo__search` is not correct - Check the Dropdown Density Sample

**Review by SisIvanova — APPROVED** [review]

**Review by simeonoff — APPROVED** [review]

**Review by damyanpetev — COMMENTED** [review]
Not sure what the `sizable()` mixin entails, however I do see a lot of size CSS vars going around. Does this mean this feature is the global CSS var-based sizing?
If so I don't see any changes to the old density handling. Guessing all the magic is handled through the variant classes (nice one), but was sort of expecting component change as well.

In any case, some changelog notes of what this feature is would be nice.

File: projects/igniteui-angular/src/lib/core/styles/components/paginator/_paginator-theme.scss [hunk 1]
```diff
@@ -56,10 +56,14 @@
 @mixin paginator($theme) {
     @include css-vars($theme);
 
-    $paginator-padding: (
-        comfortable: 0 rem(24px),
-        cosy: 0 rem(16px),
-        compact: 0 rem(12px)
+    igx-paginator {
```
[the remark this answers is not in the document]
Comment (simeonoff) [reply projects/igniteui-angular/src/lib/core/styles/components/paginator/_paginator-theme.scss]: Fixed in [refactor(paginator): use placeholder selector for sizable](https://github.com/IgniteUI/igniteui-angular/pull/12471/commits/86eb0a6640ad3834fe0778c33159a524eb799ad8)

**Review by simeonoff — APPROVED** [review]

**Pull request merged**